### PR TITLE
in-dexter:0.4.2

### PR DESCRIPTION
**Pull request by RolfBremer**
I am submitting
- [ ] a new package
- [x] an update for a package

Description: 

* Improve internal method `as-text` to be more robust.
* tidy up sample-usage.typ.
* Bug fixed: Fix a bug where an index entry with same name as a group hides the group.
* Fixed typos in the sample-usage document.
* Support for a `display` parameter for entries. This allows the usage of complex
  content, like math expressions in the index. (based on suggestions by @lukasjuhrich)
* Also support a tuple value for display and key parameters of the entry.
* Improve internal robustness and fix some errors in the sample document.
* Fix initial parsing and returning fist letter (thanks to @lukasjuhrich, #14)
* Fix handling of trailing or multiple spaces and newlines in index entries.

### CODE REVIEW

File: packages/preview/in-dexter/0.4.2/README.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+## Importing the Component
+
+To use the index functionality, the component must be available. This
+can be achieved by importing the package `in-dexter` into the project:
+
+Add the following code to the head of the document file(s)
+that want to use the index:
+
+```typ
+  #import "@preview/in-dexter:0.3.0": *
```
Comment (elegaanz) [inline packages/preview/in-dexter/0.4.2/README.md:49]: You may want to update this import.

File: packages/preview/in-dexter/0.4.2/README.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+#index-main[Willkommen]
+```
+
+### Brief Sample Document
+
+This is a very brief sample to demonstrate how in-dexter can be used. The next chapter
+contains a more fleshed out sample.
+
+```typ
+#import "@preview/in-dexter:0.3.0": *
```
Comment (elegaanz) [inline packages/preview/in-dexter/0.4.2/README.md:141]: Same here.

**Review by elegaanz — COMMENTED** [review]
LGTM, just two imports that still use `0.3.0` in the README

**Comment by RolfBremer** [comment]
Ok, done. Thank you!

**Comment by elegaanz** [comment]
Thanks!

**Pull request merged**